src/lib.rs

```diff
@@ -136,7 +136,55 @@ pub fn select_coin_lowestlarger(
     inputs: &[OutputGroup],
     options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let mut accumulated_value: u64 = 0;
+    let mut accumulated_weight: u32 = 0;
+    let mut selected_inputs: Vec<usize> = Vec::new();
+    let mut estimated_fees: u64 = 0;
+    let target = options.target_value + options.min_drain_value;
+
+    let mut sorted_inputs: Vec<_> = inputs.iter().enumerate().collect();
+    sorted_inputs.sort_by_key(|(_, input)| effective_value(input, &options));
+
+    let mut index = sorted_inputs.partition_point(|(_, input)| {
+        input.value <= (target + calculate_fee(input.weight, options.target_feerate))
+    });
+
+    for (idx, input) in sorted_inputs.iter().take(index).rev() {
+        if accumulated_value >= (target + estimated_fees.max(options.min_absolute_fee)) {
+            break;
+        }
+
+        accumulated_value += input.value;
+        accumulated_weight += input.weight;
+        estimated_fees = calculate_fee(accumulated_weight, options.target_feerate);
+        selected_inputs.push(*idx);
+    }
+
+    for (idx, input) in sorted_inputs.iter().skip(index) {
```

Comment:
We have to look for larger outputs only if we don't have adequate smaller ones.
So add that if condition before this for loop